### PR TITLE
feat: setup tests

### PR DESCRIPTION
**Pull request by Xstoudi**
We take advantage of Github Actions services to directly run a mongo container without separately installing docker on the GitHub runner.

At the moment, the lint and tests are run on node.js 12.x and 14.x as asked on #7 . Mongo instance is latest. We can setup its version too as we want.

### CODE REVIEW

**Comment by Xstoudi** [comment]
Editing according to what we discussed with @targos .
Ready to merge on my side.

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -2,38 +2,42 @@ name: Node.js CI
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Run tests
+        run: |
+          npm install
+          npm run test
```
Comment (targos) [inline .github/workflows/nodejs.yml:41]:
```suggestion
          npm run test-only
```

File: package.json [hunk 2]
```diff
@@ -42,9 +42,8 @@
     "eslint": "eslint commands providers src --ext ts --cache",
     "eslint-fix": "npm run eslint -- --fix",
     "prepublishOnly": "npm run tsc",
-    "test": "npm run eslint",
-    "test-coverage": "npm run test-only -- --coverage",
-    "test-only": "jest",
+    "test": "jest",
+    "test-coverage": "jest --coverage",
```
Comment (targos) [inline package.json:46]: please keep the "test-only" script and update "test" to be: `npm run test-only && npm run eslint`

**Review by stropitek — APPROVED** [review]
SGTM

File: src/__tests__/Connection.test.ts [hunk 3]
```diff
@@ -20,6 +20,10 @@ const connectionConfig = {
 const logger = new FakeLogger(loggerConfig);
 const connection = new Connection('mongo', connectionConfig, logger);
 
+afterAll(async () => {
+  await connection.close();
+});
+
 test('try to connect with good config', async () => {
   connection.connect();
```
Comment (targos) [inline src/__tests__/Connection.test.ts:28]: Shouldn't this be in a `beforeAll` block?

**Pull request merged**